Feature: The message above the timer should reflect how much time has passed

saszaCodes opened the pull request:
#### What does this PR do?
It makes the message displayed above the timer depend on what part of a time inputted by user has already passed. The way it is accomplished is by setting `initialTime` stateful value to equal `input` provided by the user when they click START button and therefore run `startTimer()` function. `initialTime` is not changed when calling `resumeTimer()`. `ClockComment.js` is rerendered each time `timeLeft` value is changed and therefore `progress` value is recalculated.
#### Description of Task to be completed?
Add a feature making the message above the timer reflect how much time has passed
#### How should this be manually tested?
Run timer at different speeds and values and check whether the message changes at each breakpoint (0% has passed -> 25% has passed -> 50% -> 75% -> 100%)

saszaCodes committed:
made the message above timer reflect how much time has passed

saszaCodes commented on src/App.js:
```diff
@@ -1,42 +1,102 @@
+   */
+  function resumeTimer() {
+    myInterval.current = setInterval(() => {
+      setTimeLeft((curValue) => curValue - 1);
```
this warning is the same as in PR #7 and I responded to it there.

saszaCodes commented on src/Components/InputBar.js:
```diff
@@ -1,17 +1,83 @@
-        <Button className="rounded-0 bg-success border-0">START</Button>
-      </Form>
-    </Row>
+    <>
```
this warning is the same as in PR #7 and I responded to it there.

Avuidrauxs commented:
![timer-input](https://user-images.githubusercontent.com/5025923/127310686-a7e6d89d-ae49-479e-8166-ef1b28b34901.gif)
Good job overall @tepeszCodes ,

So I have a feature tweak request. Can we rather make sure the `START` button is disabled when the wrong value is input and as they are typing the wrong value it shows the error dialog?

Maybe use `onKeyUp` or `onKeyDown` event to track that?

saszaCodes committed:
implemented requested changes to when message is displayed and when button is active

saszaCodes commented:
@Avuidrauxs I implemented the tweak that you requested. To make the code more readable and easier to tweak in the future I deleted the `showWarning` state and replaced it with more general `isValid` state, than only added an additional check `input !== ''` in the condition controlling whether to display the error message. I also used `useEffect()` hook instead of `onKeyDown` event because it seemed more appropriate to track what happens in a controlled component, since every change to it triggers the state update

Avuidrauxs committed:
Merge branch 'main' of github.com:Avuidrauxs/interactive-timer-project-001 into feature/message-above-timer

Avuidrauxs commented:
![timer-input](https://user-images.githubusercontent.com/5025923/127632087-b45336e1-4ec5-4153-ad1e-acd61794dc66.gif)

Can we have the play/pause button not touch the seconds text

saszaCodes committed:
used ReactFitty package to make clock display always be the right size

saszaCodes committed:
Merge branch 'feature/message-above-timer' of https://github.com/Avuidrauxs/interactive-timer-project-001 into feature/message-above-timer

saszaCodes commented:
@Avuidrauxs ok, I used React Fitty package that I found on npm to ensure proper font sizing

Avuidrauxs committed:
Remove package-lock

Avuidrauxs committed:
Merge branch 'feature/message-above-timer' of github.com:Avuidrauxs/interactive-timer-project-001 into feature/message-above-timer
